in_session.go

```diff
@@ -27,6 +27,16 @@ type inSession struct{ loggedOn }
 func (state inSession) String() string { return "In Session" }
 
 func (state inSession) FixMsgIn(session *session, msg *Message) sessionState {
+
+	if session.CleanIncomingHotPath && IsExecutionReport(msg.rawMessage.Bytes()) {
+		if err := session.application.FromApp(msg, session.sessionID); err != nil {
```

Comment:
so we are essentially skipping all verification checks being done here? https://github.com/alpacahq/quickfix/blob/3ce9479197a8f43fa301fbe43dc3864d4b761d43/session.go#L515

I understand the goal of this PR is to avoid parsing the message, but not sure about skipping these checks. Maybe we still populate the headers? to avoid too many allocs, you can try using sync.Pool for NewMessage()

Reply:
the existing benchmark for ParseMessage does not show allocs if a Message is already inited:

```
Running tool: /opt/homebrew/bin/go test -benchmem -run=^$ -bench ^BenchmarkParseMessage$ github.com/quickfixgo/quickfix

goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkParseMessage-10    	 1695877	       694.8 ns/op	       0 B/op	       0 allocs/op
PASS
ok  	github.com/quickfixgo/quickfix	2.079s
```

Reply:
I am not sure about the theory these allocs - have you benchmarked specific code paths or looked at the profile to confirm?

I tried to test your benchmark (without the PR changes) and I noticed it failing the msg verification checks which is why you see all the improvement by skipping it.

first it's throwing `CompID problem` so you need to fix the session creation to match the msg: `SessionID{BeginString: "FIX.4.2", TargetCompID: "ISLD", SenderCompID: "LPCA"}`

then it hits `SendingTime accuracy problem` so I just commented this line to skip it for testing: https://github.com/alpacahq/quickfix/blob/3ce9479197a8f43fa301fbe43dc3864d4b761d43/session.go#L635

then the benchmark looks much better:

```
goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkIncoming
BenchmarkIncoming-10    	  219832	      5600 ns/op	    9030 B/op	      30 allocs/op
PASS
ok  	github.com/quickfixgo/quickfix	1.784s
```

some of this can be improved if you use a sync.Pool for the Message to reuse the internal allocs.

I am not sure ParseMessage is really as bad as you think with a pre-allocated Message.

Reply:
I updated the benchmark to resolve all rejects and the results are much better now: https://github.com/alpacahq/quickfix/compare/develop...bench/incoming-neal

```
❯ go test -benchmem -run=BenchmarkIncoming -bench=BenchmarkIncoming -v
goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkIncoming
BenchmarkIncoming-10    	  241856	      4902 ns/op	    8902 B/op	      27 allocs/op
PASS
ok  	github.com/quickfixgo/quickfix	2.554s
```

and using sync.Pool for the Message, allocs go down to zero:

```
goos: darwin
goarch: arm64
pkg: github.com/quickfixgo/quickfix
BenchmarkIncoming
BenchmarkIncoming-10    	  399391	      2998 ns/op	       7 B/op	       0 allocs/op
PASS
ok  	github.com/quickfixgo/quickfix	2.618s
```